### PR TITLE
Code to checkstyle

### PR DESCRIPTION
**Pull request by SarahKurek**
**Description**
Fixed checkstyle warnings: Renamed files, adapted dummy comments.
Added short description and author to each class.
Deleted unused class Model.

**Affects**
Whole app

**Notes for Reviewer**
@xSabrina check if all checkstyle warnings are removed and check if the comments fit.

### CODE REVIEW

File: BibBuddy/app/src/main/java/de/bibbuddy/Author.java [hunk 1]
```diff
@@ -38,7 +40,7 @@ public Author(String firstName, String lastName) {
   }
 
   /**
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Constructor for optional title.
    *
    * @param firstName test
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/Author.java:45]:
```suggestion
   * @param firstName first name of the author of the book
```

Need to adjust all dummy params ("test") - this is just an example

**Review by xSabrina — CHANGES_REQUESTED** [review]
You adapted the dummy comments description, but you need to adjust all param declarations in the JavaDoc-comments, those have still dummy comment.

Nice you added short descriptions and deleted the unused class Model.

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryAddShelfFragment.java [hunk 2]
```diff
@@ -44,7 +48,9 @@ private void setupInput(View view) {
   }
 
   /**
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Closes the LibraryAddShelfFragment.
+   *
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryAddShelfFragment.java:53]:
```suggestion
  
```
Remove this author declaration in a method javaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java [hunk 3]
```diff
@@ -3,19 +3,30 @@
...
+ */
 public class DataValidation {
 
-  //checks if the input string is not null
+  /**
+   * Checks if the input string is not null.
+   *
+   * @param inputStr input String of a user
+   * @return Returns true if the String is empty
+   * @author Silvia Ivanova
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java:18]:
```suggestion
```
Remove author declarations from method-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java [hunk 4]
```diff
@@ -3,19 +3,30 @@
...
   /**
-   * DUMMY COMMENT - PLEASE ADJUST.
-   * //checks for special characters that are not latin letters or numbers
+   * Checks for special characters that are not latin letters or numbers.
    *
-   * @param inputStr test
-   * @return test
+   * @param inputStr input String of a user
+   * @return Returns true if the String has special characters
+   * @author Silvia Ivanova
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java:29]:
```suggestion
```
```suggestion
```
Remove author declarations from method-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java [hunk 5]
```diff
@@ -24,10 +35,11 @@ public static boolean hasSpecialChars(String inputStr) {
...
 
   /**
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Checks if the String is a valid ISBN13 String.
    *
-   * @param isbnStr test
-   * @return test
+   * @param isbnStr ISBN String of the input
+   * @return Returns true if the String is a valid ISBN13 String
+   * @author Silvia Ivanova
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java:42]:
```suggestion
```
Remove author declarations from method-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java [hunk 6]
```diff
@@ -64,10 +76,11 @@ public static boolean validateIsbn13(String isbnStr) {
...
 
   /**
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Checks if the String is a valid ISBN10 String.
    *
-   * @param isbnStr test
-   * @return test
+   * @param isbnStr ISBN String of the input
+   * @return Returns true if the String is a valid ISBN10 String
+   * @author Silvia Ivanova
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java:83]:
```suggestion
```
Remove author declarations from method-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java [hunk 7]
```diff
@@ -104,10 +117,11 @@ public static boolean validateIsbn10(String isbnStr) {
...
 
   /**
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Checks if the String is a valid ISBN10 or ISBN13 String.
    *
-   * @param isbnStr test
-   * @return test
+   * @param isbnStr ISBN String of the input
+   * @return Returns true if the String is a valid ISBN10 or ISBN13 String
+   * @author Silvia Ivanova
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/DataValidation.java:124]:
```suggestion
```
Remove author declarations from method-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java [hunk 8]
```diff
@@ -10,18 +10,26 @@
...
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Constructor of the BookRecyclerViewAdapter.
    *
-   * @param bookList test
-   * @param listener test
-   * @param context  test
+   * @param bookList bookList of the current books
+   * @param listener listener for the interface and callback of the books
+   * @param context  context is required for the BookRecyclerViewAdapter texts (getNoteString)
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java:32]:
```suggestion
```
Remove author declarations from constructor-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java [hunk 9]
```diff
@@ -83,9 +92,10 @@ public int getItemCount() {
     private final TextView textNoteCount;
 
     /**
-     * DUMMY COMMENT - PLEASE ADJUST.
+     * Custom ViewHolder constructor to setup its basic view.
      *
-     * @param itemView test
+     * @param itemView View of the BookRecyclerView-item.
+     * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookRecyclerViewAdapter.java:98]:
```suggestion
```
Remove author declarations from constructor-JavaDoc.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookNotesRecyclerViewAdapter.java [hunk 10]
```diff
@@ -58,7 +64,7 @@ public int getItemCount() {
     private final ImageView imageView;
 
     /**
-     * DUMMY COMMENT - PLEASE ADJUST.
+     * Custom ViewHolder constructor to setup basic view.
      *
      * @param itemView test
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookNotesRecyclerViewAdapter.java:69]: @param needs to be adjusted

File: BibBuddy/app/src/main/java/de/bibbuddy/BookModel.java [hunk 11]
```diff
@@ -4,12 +4,26 @@
...
   private final Long shelfId;
 
   private List<BookItem> bookList;
 
+  /**
+   * Constructor for a BookModel.
+   *
+   * @param context context for the BookModel
+   * @param shelfId shelfId of the selected book
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookModel.java:24]:
```suggestion
```
Remove author-declaration from constructor javaDoc-Comment.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookModel.java [hunk 12]
```diff
@@ -34,13 +48,27 @@ private String convertAuthorListToString(List<Author> authorList) {
     return authors.toString();
   }
 
+  /**
+   * Add a new book to the bookList and database.
+   *
+   * @param book       book data for the database and bookList
+   * @param authorList authorList of the new book
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookModel.java:56]:
```suggestion
```
Remove author-declaration from method javaDoc-Comment.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookModel.java [hunk 13]
```diff
@@ -34,13 +48,27 @@ private String convertAuthorListToString(List<Author> authorList) {
...
     bookList
         .add(new BookItem(book.getTitle(), book.getId(), shelfId, book.getPubYear(), authors, 0));
   }
 
+  /**
+   * Get the bookList of the current shelfId.
+   *
+   * @param shelfId shelfId of the given shelf
+   * @return Returns the bookList of the current shelfId
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookModel.java:70]:
```suggestion
```
Remove author-declaration from method javaDoc-Comment.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookItem.java [hunk 14]
```diff
@@ -9,6 +15,17 @@ public BookItem(String title, Long id) {
...
+  /**
+   * Constructor for a BookItem.
+   *
+   * @param title     title of the book
+   * @param id        id of the book
+   * @param shelfId   shelfId of the book
+   * @param year      published year of the book
+   * @param authors   authors of the book
+   * @param noteCount total number of notes in the book
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookItem.java:27]:
```suggestion
```
Remove author-declaration from method javaDoc-Comment.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookDao.java [hunk 15]
```diff
@@ -63,14 +68,8 @@ public void create(Book book, List<Author> authorList, Long shelfId) {
...
 
-
-  /**
-   * DUMMY COMMENT - PLEASE ADJUST.
-   * // get single book entry
-   *
-   * @param id test
-   * @return test
-   */
+  // get a single book entry by id
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookDao.java:71]: Why did you replace the JavaDoc comment by a basic //?

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java [hunk 16]
```diff
@@ -18,11 +24,12 @@
...
-   * DUMMY COMMENT - PLEASE ADJUST.
+   * Constructor of the BookRecyclerViewAdapter.
    *
-   * @param libraryList test
-   * @param listener    test
-   * @param context     test
+   * @param libraryList libraryList of the current shelves
+   * @param listener    listener for the interface and callback of the shelves
+   * @param context     context is required for the LibraryRecyclerViewAdapter texts (getBookString)
+   * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java:32]:
```suggestion
```
Remove author-declaration from constructor javaDoc-Comment.

File: BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java [hunk 17]
```diff
@@ -103,9 +110,10 @@ public int getItemCount() {
     private final TextView textNoteCount;
 
     /**
-     * DUMMY COMMENT - PLEASE ADJUST.
+     * The LibraryViewHolder describes a shelf item view and metadata about its place
+     * within the RecyclerView of the LibraryFragment.
      *
-     * @param itemView test
+     * @author Claudia Schönherr
```
Comment (xSabrina) [inline BibBuddy/app/src/main/java/de/bibbuddy/LibraryRecyclerViewAdapter.java:116]:
```suggestion
```
Remove author-declaration from constructor javaDoc-Comment.

**Comment by xSabrina** [comment]
There are 2 checkstyle warnings left to resolve (besides from the ones that are caused by legacy test-classes):

- BibBuddy/app/build/generated/source/buildConfig/release/de/bibbuddy/BuildConfig.java|4| 'package' should be separated from previous statement.

- BibBuddy/app/build/generated/source/buildConfig/debug/de/bibbuddy/BuildConfig.java|4| 'package' should be separated from previous statement.

There are comments above the package declaration - a blank line needs to be inserted between those.

**Comment by SarahKurek** [comment]
> There are 2 checkstyle warnings left to resolve (besides from the ones that are caused by legacy test-classes):
> 
>     * BibBuddy/app/build/generated/source/buildConfig/release/de/bibbuddy/BuildConfig.java|4| 'package' should be separated from previous statement.
> 
>     * BibBuddy/app/build/generated/source/buildConfig/debug/de/bibbuddy/BuildConfig.java|4| 'package' should be separated from previous statement.
> 
> 
> There are comments above the package declaration - a blank line needs to be inserted between those.

BuildConfig file is automatically generated and could not be changed.

**Review by xSabrina — APPROVED** [review]
Params are declared correctly and legacy author-declarations have been removed.

The changes can be merged now.

**Comment by xSabrina** [comment]
> 
> 
> > There are 2 checkstyle warnings left to resolve (besides from the ones that are caused by legacy test-classes):
> > ```
> > * BibBuddy/app/build/generated/source/buildConfig/release/de/bibbuddy/BuildConfig.java|4| 'package' should be separated from previous statement.
> > 
> > * BibBuddy/app/build/generated/source/buildConfig/debug/de/bibbuddy/BuildConfig.java|4| 'package' should be separated from previous statement.
> > ```
> > 
> > 
> > There are comments above the package declaration - a blank line needs to be inserted between those.
> 
> BuildConfig file is automatically generated and could not be changed.

Strange that checkstyle will not approve it - we will leave it as it is then.

**Pull request merged**